packages/flutter/test/widgets/two_dimensional_viewport_test.dart

```diff
@@ -2713,6 +2713,26 @@ void main() {
       });
     });
 
+    testWidgets('Does not throw when no child is laid out',
+        (WidgetTester tester) async {
+      final TwoDimensionalChildBuilderDelegate delegate =
+          TwoDimensionalChildBuilderDelegate(
+              maxXIndex: 50,
+              maxYIndex: 50,
+              addAutomaticKeepAlives: false,
+              addRepaintBoundaries: false,
+              builder: (BuildContext context, ChildVicinity vicinity) {
+                if (vicinity.xIndex > 10) {
+                  return const SizedBox.square(dimension: 200);
+                }
+                return null;
+              });
+      addTearDown(delegate.dispose);
+
+      await tester.pumpWidget(simpleBuilderTest(delegate: delegate));
+      expect(tester.takeException(), isNull);
+    }, variant: TargetPlatformVariant.all());
```

Comment:
Nit: I don't think you need to use all platforms here since the behavior should be the same.